src/strategies/get-xkcd/extractor.mjs

```diff
@@ -26,26 +36,49 @@ export function init(start = 1) {
 export function update(message) {
   // TODO: There's a bug in neume-network/extraction-worker that doesn't
   // allow us to get back the error: https://github.com/neume-network/extraction-worker/issues/34
-  if (message.error && message.error.includes("404 Not Found")) {
+  if (message.error) {
+    // handle the error
+    log(message.error);
+    // stop the crawler if necessary
     return {
-      write: null,
-      messages: [],
+      type: "exit",
+      version: "1.0",
     };
   }
+
+  const data = message.results;
+  if (!validate(data)) {
```

Comment:
great. I commented elsewhere that I dislike shutting down the entire crawl, but now seeing that the implementation uses json schema, this type of shutdown is appropriate! awesome work!

Reply:
Does the issue https://github.com/neume-network/strategies/issues/156 mean to do what get-xkcd has done in this branch?